lib/ain-evm/src/handler.rs

```diff
@@ -169,12 +175,15 @@ impl Handlers {
             block.header.number,
         );
 
+        // calculate base fee
+        let base_fee = self.block.calculate_base_fee(parent_hash);
+
         if update_state {
             debug!(
                 "[finalize_block] Finalizing block number {:#x}, state_root {:#x}",
                 block.header.number, block.header.state_root
             );
```

Comment:
```suggestion
        if update_state {
            debug!(
                "[finalize_block] Finalizing block number {:#x}, state_root {:#x}",
                block.header.number, block.header.state_root
            );
        // calculate base fee
        let base_fee = self.block.calculate_base_fee(parent_hash);
```

Only used when state is updated